modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+|===
+
+[discrete]
+== `useK8sWatchResources`
+
+Hook that retrieves the k8s resources along with their respective status for loaded and error. It returns a map where keys are as provided in initResouces and value has three properties data, loaded and error.
+
+[source,ts]
+----
+const Component: React.FC = () => {
+  const watchResources = {
+        'deployment': {...},
+        'pod': {...}
+        ...
+      }
+  const {deployment, pod} = useK8sWatchResources(watchResources)
+  return ...
+}
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`initResources` |resources need to be watched as key-value pair,
+wherein key will be unique to resource and value will be options needed
+to watch for the respective resource.
+|===
+
+[discrete]
+== `consoleFetch`
+
+A custom wrapper around `fetch` that adds console specific headers and allows for retries and timeouts.It also validates the response status code and throws appropriate error or logs out the user if required. It returns a promise that resolves to the response.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`url` |The URL to fetch
+|`options` |The options to pass to fetch
+|`timeout` |The timeout in milliseconds
+|===
+
+[discrete]
+== `consoleFetchJSON`
+
+A custom wrapper around `fetch` that adds console specific headers and allows for retries and timeouts. It also validates the response status code and throws appropriate error or logs out the user if required. It returns the response as a JSON object. Uses `consoleFetch` internally. It returns a promise that resolves to the response as JSON object.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`url` |The URL to fetch
+
+|`method` |The HTTP method to use. Defaults to GET
+
+|`options` |The options to pass to fetch
+
+|`timeout` |The timeout in milliseconds
+
+|`cluster` |The name of the cluster to make the request to. Defaults to
+the active cluster the user has selected
+|===
+
+[discrete]
+== `consoleFetchText`
+
+A custom wrapper around `fetch` that adds console specific headers and allows for retries and timeouts. It also validates the response status code and throws appropriate error or logs out the user if required. It returns the response as a text. Uses `consoleFetch` internally. It returns a promise that resolves to the response as text.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`url` |The URL to fetch
+
+|`options` |The options to pass to fetch
+
+|`timeout` |The timeout in milliseconds
+
+|`cluster` |The name of the cluster to make the request to. Defaults to
+the active cluster the user has selected
+|===
+
+[discrete]
+== `getConsoleRequestHeaders`
+
+A function that creates impersonation and multicluster related headers for API requests using current redux state. it returns an object containing the appropriate impersonation and clustr requst headers, based on redux state.
```

Comment:
s/impersonation and multicluster related headers/impersonation- and multicluster-related headers